crates/rspack_plugin_javascript/src/plugin/impl_plugin_for_js_plugin.rs

```diff
@@ -225,6 +225,12 @@ impl Plugin for JsPlugin {
   ) -> PluginProcessAssetsOutput {
     let compilation = args.compilation;
     let minify_options = &compilation.options.builtins.minify_options;
+    let is_module = compilation
+      .options
+      .output
+      .library
+      .as_ref()
+      .is_some_and(|library| library.library_type == "module");
```

Comment:
@zackarychapple @suxin2017 I just stumbled upon this MR because I have a similar issue with output.library.type = 'system'.  In 2 of 3 cases it fails due to the error: ^^^^^^ 'import', and 'export' cannot be used outside of module code. Should we add systemjs as another type here?